titiler/endpoints/factory.py

```diff
@@ -41,7 +45,6 @@
 from starlette.requests import Request
 from starlette.responses import Response
 
-default_readers_type = Union[Type[BaseReader], Type[MultiBaseReader]]
```

Comment:
Because it wasn't true that TilerFactories was supporting `MultiBaseReader`. With this PR we add factories especially for the MultiBaseReader and MultiBandReader.

Reply:
the less unions the better!